File: Backend/BandFounder/BandFounder.Domain/Entities/MusicianSlot.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BandFounder.Domain.Entities;

public class MusicianSlot : Entity
{ 
    [Key]
    public Guid Id { get; set; }

    public virtual MusicianRole Role { get; set; } = null!;

    [Column(TypeName = "text")]
    public virtual SlotStatus Status { get; set; }

    public virtual Guid ListingId { get; set; }
    public virtual Listing Listing { get; set; } = null!;
}

public enum SlotStatus
{
    Available,
    Filled
}

File: Frontend/BandFounder/src/types/MusicianSlot.ts
export interface MusicianSlot {
    id: string;
    role: string;
    status: SlotType;
}

export enum SlotType {
    Available = 'Available',
    Filled = 'Filled'
}

export function castToSlotType(value: string): SlotType | null {
    return isSlotType(value) ? (value as SlotType) : null;
}

function isSlotType(value: string): value is SlotType {
    return Object.values(SlotType).includes(value as SlotType);
}